public/app/listeners/executePythonAlgorithm.js

```diff
@@ -33,15 +34,23 @@ const executePythonAlgorithm = (mainWindow, db) => (event, { datasetId }) => {
   process.on('close', (code) => {
     if (code !== 0) {
       logger.error(`python process exited with code ${code}`);
+
+      // delete tmp file
+      if (fs.existsSync(outputPath)) {
```

Comment:
What do you think about moving this conditional block after the if-else block? (and remove line 50-51)